File: javascript/leetcode_8.js
var myAtoi = function(s) {
    s = s.trim();

    let number = 0;

    let negative = s[0] === "-"? true : false
    let start = s[0] === "-" || s[0] === "+" ? 1 : 0

    for (let i = start; i < s.length; i++){
        let code = s.charCodeAt(i);
        if ( code >= 48 &&  code <= 57){
            number = number*10 + (code-48)
        //if (s[i] in ['0','1','2','3','4','5','6','7','8','9']){
        //    number = number*10 + parseInt(s[i]);
        } else {
            break;
        }
    }

    number  = negative ? -1*number : number; 
    number = number < -2147483648 ? -2147483648 : number;
    number = number > 2147483647 ? 2147483647 : number;

    
    return number;
};
    

let val = myAtoi("123")
console.log(val, typeof val);
val = myAtoi("-123")
console.log(val, typeof val);
val = myAtoi("+123")
console.log(val, typeof val);
val = myAtoi("0")
console.log(val, typeof val);
val = myAtoi("-91283472332")
console.log(val, typeof val);
//console.log(myAtoi("+123"));
//console.log(myAtoi("-1233   "));

File: python/leetcode_8.py
from tkinter.tix import INTEGER


@profile
def myAtoi( s):
    s = s.lstrip()
    if len(s) == 0:
        return 0


    if s[0] == "-":
        negative,start = True,1
    elif s[0] == "+":
        negative,start = False,1
    else:
        negative,start = False,0

    number = 0

    for c in range(start,len(s)):
        code = ord(s[c])
        if code >= 48 and code <= 57:
            number = number*10 + (code - 48)
        else:
            break
        
        
    number = -1*number if negative else number

    number = -2147483648 if number < -2147483648 else number
    number = 2147483647 if number > 2147483647 else number

    return number

if __name__ == '__main__':
    print(myAtoi("12345678"))
    print(myAtoi("+345"))
    print(myAtoi("-12345678"))
    print(myAtoi("-394589512345678"))
    print(myAtoi("+12345678"))
    print(myAtoi("345678"))
    print(myAtoi("     345678"))
